app/src/main/java/com/firebase/samples/logindemo/MainActivity.java

```diff
@@ -123,14 +130,16 @@ protected void onCreate(Bundle savedInstanceState) {
         /* *************************************
          *              FACEBOOK               *
          ***************************************/
-        /* Load the Facebook login button and set up the session callback */
+        /* Load the Facebook login button and set up the tracker to monitor access token changes */
+        mFacebookCallbackManager = CallbackManager.Factory.create();
         mFacebookLoginButton = (LoginButton) findViewById(R.id.login_with_facebook);
-        mFacebookLoginButton.setSessionStatusCallback(new Session.StatusCallback() {
+        mFacebookAccessTokenTracker = new AccessTokenTracker() {
```

Comment:
Wow, you're really paying attention. I got most of this from the Scrumptious sample in the Facebook SDK, which -guess what- also doesn't call startTracking. Given that tracking works, it indeed seems to track automatically. The Facebook docs are also quite terse on its intended use: https://developers.facebook.com/docs/reference/android/current/class/AccessTokenTracker/ "This class can be extended to receive notifications of access token changes. The AccessTokenTracker.stopTracking() method should be called in the onDestroy() method of the receiving Activity or Fragment."